src/libraries/System.Net.Mail/tests/Functional/SmtpClientTest.cs

```diff
@@ -349,60 +345,38 @@ public void TestZeroTimeout()
-                Thread t = new Thread(server.Run);
-                t.Start();
-                await client.SendMailAsync(msg).TimeoutAfter((int)TimeSpan.FromSeconds(30).TotalMilliseconds);
-                t.Join();
-
-                Assert.Equal("<foo@example.com>", server.MailFrom);
-                Assert.Equal("<bar@example.com>", server.MailTo);
-                Assert.Equal("hello", server.Subject);
-                Assert.Equal(body ?? "", server.Body);
-                Assert.Equal(clientDomain, server.ClientDomain);
-            }
-            finally
-            {
-                server.Stop();
-            }
+            await client.SendMailAsync(msg).TimeoutAfter((int)TimeSpan.FromSeconds(30).TotalMilliseconds);
+
+            Assert.Equal("<foo@example.com>", server.MailFrom);
+            Assert.Equal("<bar@example.com>", server.MailTo);
+            Assert.Equal("hello", server.Message.Subject);
+            Assert.Equal(body ?? "", server.Message.Body);
+            Assert.Equal(GetClientDomain(), server.ClientDomain);
         }
 
         [Fact]
+        [ActiveIssue(28961)]
+        [PlatformSpecific(TestPlatforms.Windows)] // NTLM support required
         public async Task TestCredentialsCopyInAsyncContext()
         {
-            SmtpServer server = new SmtpServer();
-            SmtpClient client = new SmtpClient("localhost", server.EndPoint.Port);
+            using var server = new MockSmtpServer();
+            using SmtpClient client = server.CreateClient();
             MailMessage msg = new MailMessage("foo@example.com", "bar@example.com", "hello", "howdydoo");
-            string clientDomain = IPGlobalProperties.GetIPGlobalProperties().HostName.Trim().ToLower();
 
             CredentialCache cache = new CredentialCache();
-            cache.Add("localhost", server.EndPoint.Port, "NTLM", CredentialCache.DefaultNetworkCredentials);
+            cache.Add("localhost", server.Port, "NTLM", CredentialCache.DefaultNetworkCredentials);
 
             client.Credentials = cache;
 
-            try
-            {
-                Thread t = new Thread(server.Run);
-                t.Start();
-                await client.SendMailAsync(msg);
-                t.Join();
-
-                Assert.Equal("<foo@example.com>", server.MailFrom);
-                Assert.Equal("<bar@example.com>", server.MailTo);
-                Assert.Equal("hello", server.Subject);
-                Assert.Equal("howdydoo", server.Body);
-                Assert.Equal(clientDomain, server.ClientDomain);
-            }
-            finally
-            {
-                server.Stop();
-            }
+            // The mock server doesn't actually understand NTLM, but still advertises support for it
+            server.AdvertiseNtlmAuthSupport = true;
+            await Assert.ThrowsAsync<SmtpException>(async () => await client.SendMailAsync(msg));
+
+            Assert.Equal("ntlm", server.AuthMethodUsed, StringComparer.OrdinalIgnoreCase);
```

Comment:
"ntlm" -> "NTLM".  Best-practice is that the server advertises the protocols as per RFC specs.  'NTLM' is specified as uppercase token.

Reply:
The test server does, SmtpClient announces with "AUTH ntlm" or "AUTH login" tho